src/BenchmarksServer/Startup.cs

```diff
@@ -217,6 +221,28 @@ public static int Main(string[] args)
                                 Debug.Assert(process == null);
                                 process = StartProcess(hostname, database, sqlConnectionString, Path.Combine(tempDir, benchmarksDir),
                                     job, dotnetHome);
+                                
+                                if (timer != null)
+                                {
+                                    timer.Dispose();
+                                }
+
+                                DateTime lastMonitorTime = DateTime.UtcNow;
+                                var oldCPUTime = new TimeSpan(0);
+
+                                timer = new Timer(_ => 
+                                {
+
+                                    var newCPUTime = process.TotalProcessorTime;
+                                    var ellapsed = DateTime.UtcNow.Subtract(lastMonitorTime).TotalMilliseconds;
+                                    var cpu =  Math.Round((newCPUTime - oldCPUTime).TotalMilliseconds / (Environment.ProcessorCount *  ellapsed) * 100);
+                                    lastMonitorTime = DateTime.UtcNow;
+                                    oldCPUTime = newCPUTime;
+
+                                    job.WorkingSets.Add(process.WorkingSet64);
+                                    job.Cpus.Add(cpu);
+
+                                }, null, TimeSpan.FromTicks(0), TimeSpan.FromSeconds(1));
```

Comment:
Is this accurate enough? The lambda will execute after at least a second, perhaps throwing off averages of `WorkingSets` and `Cpus` values.